src/MultiphaseStaggeredStokesBlockOperator.cpp

```diff
@@ -0,0 +1,598 @@
+                un_idx, P_idx, d_un_bc_coefs, d_new_time, d_homogeneous_bc);
+            d_bc_helper->enforceNormalVelocityBoundaryConditions(
+                us_idx, P_idx, d_us_bc_coefs, d_new_time, d_homogeneous_bc);
+        }
+        apply(*x, *b);
+        y.subtract(Pointer<SAMRAIVectorReal<NDIM, double>>(&y, false), b);
+        x->freeVectorComponents();
+        b->freeVectorComponents();
+    }
+    const bool homogeneous_bc = true;
+    if (d_bc_helper)
+    {
+        const int un_idx = y.getComponentDescriptorIndex(0);
+        const int us_idx = y.getComponentDescriptorIndex(1);
+        const int P_idx = y.getComponentDescriptorIndex(2);
+        d_bc_helper->enforceNormalVelocityBoundaryConditions(un_idx, P_idx, d_un_bc_coefs, d_new_time, homogeneous_bc);
+        d_bc_helper->enforceNormalVelocityBoundaryConditions(us_idx, P_idx, d_us_bc_coefs, d_new_time, homogeneous_bc);
+    }
+    return;
+} // modifyRhsForBcs
+
+void
+MultiphaseStaggeredStokesBlockOperator::imposeSolBcs(SAMRAIVectorReal<NDIM, double>& u)
+{
+    if (d_bc_helper)
+    {
+        const int un_idx = u.getComponentDescriptorIndex(0);
+        const int us_idx = u.getComponentDescriptorIndex(1);
+        const int P_idx = u.getComponentDescriptorIndex(2);
+        d_bc_helper->enforceNormalVelocityBoundaryConditions(
+            un_idx, P_idx, d_un_bc_coefs, d_new_time, d_homogeneous_bc);
+        d_bc_helper->enforceNormalVelocityBoundaryConditions(
+            us_idx, P_idx, d_us_bc_coefs, d_new_time, d_homogeneous_bc);
+    }
+    return;
+} // imposeSolBcs
+
+void
+MultiphaseStaggeredStokesBlockOperator::applySpecialized(const int A_un_idx,
+                                                         const int A_us_idx,
+                                                         const int un_idx,
+                                                         const int us_idx,
+                                                         const int thn_idx)
+{
+    // Interpolate and synchronize volume fraction
+    d_hier_math_ops->interp(
+        d_nc_scr_idx, d_nc_scr_var, true, thn_idx, Pointer<CellVariable<NDIM, double>>(nullptr), nullptr, d_new_time);
+    // Interpolate to cell sides
+    convert_to_ndim_cc(d_cc_ndim_idx, thn_idx, d_hierarchy);
+    d_hier_math_ops->interp(d_sc_scr_idx, d_sc_scr_var, true, d_cc_ndim_idx, d_cc_ndim_var, nullptr, d_new_time);
+
+    // Compute the forces on momentum.
+    for (int ln = 0; ln <= d_hierarchy->getFinestLevelNumber(); ++ln)
+    {
+        Pointer<PatchLevel<NDIM>> level = d_hierarchy->getPatchLevel(ln);
+        for (PatchLevel<NDIM>::Iterator p(level); p; p++)
+        {
+            Pointer<Patch<NDIM>> patch = level->getPatch(p());
+            // if (d_params.isVariableDrag())
+            //     //accumulateMomentumWithoutPressureOnPatchVariableDrag(
+            //     //    patch, A_un_idx, A_us_idx, un_idx, us_idx, thn_idx, d_params, d_C, d_D_u);
+            // else
```

Comment:
What needs to be done to get variable drag to work?

Reply:
Just needs to be typed up. I was just focusing on constant coefficient as we discussed. I can add this if it's necessary.